tests/test_lab_dev/test_wires.py

```diff
@@ -27,154 +26,74 @@ class TestWires:
     Tests for the Wires class.
     """
 
-    @pytest.mark.parametrize("modes_out_bra", [[0], [1, 2]])
-    @pytest.mark.parametrize("modes_in_bra", [None, [1], [2, 3]])
-    @pytest.mark.parametrize("modes_out_ket", [None, [2], [3, 4]])
-    @pytest.mark.parametrize("modes_in_ket", [None, [3], [4, 5]])
-    def test_init(self, modes_out_bra, modes_in_bra, modes_out_ket, modes_in_ket):
-        w = Wires(modes_out_bra, modes_in_bra, modes_out_ket, modes_in_ket)
-
-        modes_out_bra = modes_out_bra or []
-        modes_in_bra = modes_in_bra or []
-        modes_out_ket = modes_out_ket or []
-        modes_in_ket = modes_in_ket or []
-        modes = list(
-            set(modes_out_bra) | set(modes_in_bra) | set(modes_out_ket) | set(modes_in_ket)
-        )
-
-        assert w.modes == modes
-        assert w.output.bra.modes == modes_out_bra
-        assert w.input.bra.modes == modes_in_bra
-        assert w.output.ket.modes == modes_out_ket
-        assert w.input.ket.modes == modes_in_ket
-
     def test_args(self):
-        w = Wires([0], [1], [2], [3])
-        assert w._args() == ((0,), (1,), (2,), (3,))
-
-    def test_from_data(self):
-        id_array = [[1, 2, 0, 0], [0, 3, 4, 0], [0, 0, 5, 0], [0, 0, 0, 6]]
-        modes = [5, 6, 7, 8]
-        w = Wires._from_data(id_array, modes)
-
-        assert np.allclose(w.id_array, id_array)
-        assert w.ids == [1, 2, 3, 4, 5, 6]
-        assert w.modes == modes
-
-    def test_view(self):
-        w = Wires([0], [0], [0], [0])
-        assert set(w.ids) == set(w._view().ids)
-
-    def test_view_can_edit_original(self):
-        w = Wires([0], [0], [0], [0])
-        w._view().ids = [9, 99, 999, 9999]
-        assert w.ids == [9, 99, 999, 9999]
+        w = Wires({0, 1, 2}, {3, 4, 5}, {6, 7}, {8})
+        assert w.args == ({0, 1, 2}, {3, 4, 5}, {6, 7}, {8})
 
     def test_wire_subsets(self):
-        w = Wires([0], [1], [2], [3])
-        assert w.output.bra.modes == [0]
-        assert w.input.bra.modes == [1]
-        assert w.output.ket.modes == [2]
-        assert w.input.ket.modes == [3]
-
-        w = Wires([10], [11], [12], [13])
-        assert w[10].ids == w.output.bra.ids
-        assert w[11].ids == w.input.bra.ids
-        assert w[12].ids == w.output.ket.ids
-        assert w[13].ids == w.input.ket.ids
-
-    def test_id_array(self):
-        w = Wires([0, 1], [2], [3, 4, 5], [6])
-        assert w.id_array.shape == (7, 4)
-
-    def test_ids(self):
-        w = Wires([0, 1], [2], [3, 4, 5], [6])
-
-        assert w.output.bra.ids == w.ids[:2]
-        assert w.input.bra.ids == [w.ids[2]]
-        assert w.output.ket.ids == w.ids[3:6]
-        assert w.input.ket.ids == [w.ids[-1]]
-
-    def test_ids_setter(self):
```

Comment:
@ziofil Could you write a test to show how one can set the ids?

- Why it's useful to set the `id`s:
To connect 2 components while constructing the graph, which is useful to either validate the user-provided path (simulator) or to find the optimal path

- How we do this now:
By manually changing the ids of the wires that contract